frontend/packages/console-app/src/hooks/useCSPVioliationDetector.tsx

```diff
@@ -108,28 +98,35 @@ const useCSPViolationReporter: UseCSPVilationReporter = () => {
     [],
   );
 
-  return React.useCallback(
-    (pluginName, event) => {
+  const reportViolation = React.useCallback(
+    (event) => {
       // eslint-disable-next-line no-console
       console.warn('Content Security Policy violation detected', event);
+
+      // Attempt to infer Console plugin name from SecurityPolicyViolation event
+      const pluginName =
+        getPluginNameFromResourceURL(event.blockedURI) ||
+        getPluginNameFromResourceURL(event.sourceFile);
+
       const existingRecords = getRecords();
-      const latestOccurrance = {
+      const newRecord = {
         ...newCSPViolationReport(pluginName, event),
         timestamp: Date.now(),
       };
-      const updatedRecords = updateRecords(existingRecords, latestOccurrance);
-      const isNewOccurrance = updatedRecords.length > existingRecords.length;
-      const shouldNotify = isNewOccurrance && process.env.NODE_ENV === 'production';
+      const updatedRecords = updateRecords(existingRecords, newRecord);
+      const isNewOccurrence = updatedRecords.length > existingRecords.length;
+      const isProduction = process.env.NODE_ENV === 'production';
 
       window.localStorage.setItem(LOCAL_STORAGE_CSP_VIOLATIONS_KEY, JSON.stringify(updatedRecords));
 
-      if (shouldNotify) {
-        fireTelemetryEvent('CSPViolation', latestOccurrance);
+      if (isNewOccurrence && isProduction) {
+        fireTelemetryEvent('CSPViolation', newRecord);
       }
 
       if (pluginName) {
         const pluginInfo = pluginStore.findDynamicPluginInfo(pluginName);
         const validPlugin = !!pluginInfo;
+        const pluginIsLoaded = validPlugin && pluginInfo.status === 'Loaded';
```

Comment:
instead of `Loaded` this could be `LoadedDynamicPluginInfo.status` to reduce duplication

Reply:
`LoadedDynamicPluginInfo` is a type def, we can't use it or it's properties as JS values.